core/src/main/java/org/mule/exception/TemplateMessagingExceptionStrategy.java

```diff
@@ -45,7 +47,7 @@ final public MuleEvent handleException(Exception exception, MuleEvent event)
             event.getMessage().setExceptionPayload(new DefaultExceptionPayload(exception));
 
             // MULE-8551 Still need to add support for non-blocking components in excepton strategies.
-            if(event.isAllowNonBlocking())
+            if(nonBlocking)
```

Comment:
Please add a unit test for this as there is no test in CE failing because of this change